packages/workspace/src/affine/sync.ts

```diff
@@ -20,12 +18,11 @@ const downloadRecursive = async (
   const binary = await downloadBinaryFromCloud(rootGuid, doc.guid);
   if (typeof binary !== 'boolean') {
     const update = new Uint8Array(binary);
-    if (rootGuid === doc.guid) {
-      // only apply the root doc
-      applyUpdate(doc, update, 'affine-cloud-service');
-    } else {
-      await pushBinary(doc.guid, update);
-    }
+    await window.apis.db.applyDocUpdate(
+      rootGuid,
+      update,
+      rootGuid === doc.guid ? undefined : doc.guid
+    );
```

Comment:
This time the data is not being written to the ydoc in the view. I am not sure if this is a good decision.